### PR TITLE
handle application logic using redux.

### PR DESCRIPTION
**Pull request by Jwhiles**
* refactor app state into redux
* implemented conversation and CTA structure
* add minor functionality, close button on CTA

### CODE REVIEW

File: src/reducers/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { combineReducers } from "redux";
+import * as actions from "../actions";
+import model from "../resources/model.js";
+
+const currentStateReducer = (state = { status: "idle", value: "" }, action) => {
```
Comment (des-des) [inline src/reducers/index.js]: Okay so your naming here is really hard to follow.

We have a reducer called currentStateReducer, exported under the name currentState.

The state tree looks like this:
```
state: {
  currentState: {
    status,
    value
  },
  ctas: ??
  messages: ??
}
```

When reviewing react I start with the reducers as normally the business logic lives, but its very hard to see what any of this means..

File: src/components/CallToAction.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+class CallToAction extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  handleClick () {
+    this.props.updateMessage({ state: "idle", value: "" })
+  }
+
+  render() {
+    let { imgSrc, ctaText, ctaURL, ctaLinkText } = this.props.value
```
Comment (des-des) [inline src/components/CallToAction.js]: Value is not a good name.

Comment (des-des) [reply src/components/CallToAction.js]: This should prob read

```js
const { imgSrc, ctaText, ctaURL, ctaLinkText } = this.props
```

File: src/containers/MenuContainer.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import { connect } from 'react-redux';
+import actions from "../actions";
+
+const mapStateToProps = (state) => {
+  return {}; // TODO - complete
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    handleEvent: (event) => {
```
Comment (des-des) [inline src/containers/MenuContainer.js]: event not needed

Comment (des-des) [reply src/containers/MenuContainer.js]: Is renaming `initaiteConversation` to `handleEvent` helpful? I expect the first is a better name..

File: src/index.js [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
 import ReactDOM from 'react-dom';
-import App from './components/App';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+
+import reducers from './reducers';
+import App from './containers/App';
 import './styles/index.css';
 
+const createStoreWithMiddleware = applyMiddleware()(createStore);
```
Comment (des-des) [inline src/index.js:10]: FYI If you are not using any middleware there is a more concise way of doing this. But happy to leave it in in case you need it later

**Review by des-des — COMMENTED** [review]
Okay, I am happy with the structure here, most of my comments are around naming...

File: src/containers/MenuContainer.js (hunk 3 above)
Comment (Jwhiles) [reply src/containers/MenuContainer.js]: We have a number of buttons, this is meant to correctly handle which ever one we click on. I've added a switch case to do so :)

File: src/containers/MenuContainer.js [hunk 5]
```diff
@@ -8,8 +8,14 @@ const mapStateToProps = (state) => {
 
 const mapDispatchToProps = (dispatch) => {
   return {
-    handleEvent: (event) => {
-      dispatch(actions.initiateConversation());
+    handleEvent: (label) => {
+      switch(label) {
+        case 'talk':
```
Comment (des-des) [inline src/containers/MenuContainer.js]: makes more sense

**Review by des-des — APPROVED** [review]

**Comment by des-des** [comment]
the name action still is not great as it is confused with redux actions..
